net/net.go

```diff
@@ -1037,15 +1044,17 @@ func (n *net) getOwnLog(id thread.ID) (info thread.LogInfo, err error) {
 			return n.store.GetLog(id, lid)
 		}
 	}
-	return info, nil
+	return info, lstore.ErrLogNotFound
```

Comment:
This felt safer (thanks to discussion with @jsign), and doesn't break anything else... though without it, we _can_ end up in a state where we have an invalid log (log that is just the zero state).